### PR TITLE
added settings cache

### PR DESCRIPTION
**Pull request by TopiSenpai**

### CODE REVIEW

**Review by TopiSenpai — COMMENTED** [review]
okay looks good to me 
have a look too @caneleex

File: src/main/java/de/kittybot/kittybot/WebService.java [hunk 1]
```diff
@@ -108,17 +107,19 @@ private static void login(Context ctx){
 
 	private static void logout(Context ctx){
 		var auth = ctx.header("Authorization");
-		if(auth != null){
```
Comment (caneleex) [inline src/main/java/de/kittybot/kittybot/WebService.java:111]: i'd keep this check as it was before, it was shorter

File: src/main/java/de/kittybot/kittybot/cache/GuildSettingsCache.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+		Database.setNSFWEnabled(guildId, enabled);
+	}
+
+	public static void setInactiveRoleId(String guildId, String roleId){
+		getGuildSettings(guildId).setInactiveRoleId(roleId);
+		Database.setInactiveRoleId(guildId, roleId);
+	}
+
+	public static void pruneCache(Guild guild){
+		SETTINGS.entrySet().removeIf(entry -> entry.getValue().getGuildId().equals(guild.getId()));
```
Comment (caneleex) [inline src/main/java/de/kittybot/kittybot/cache/GuildSettingsCache.java:161]: why not just use `SETTINGS.remove(guild.getId())` here? i believe the map key is the guild id?

File: src/main/java/de/kittybot/kittybot/objects/GuildSettings.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
+package de.kittybot.kittybot.objects;
+
+import de.kittybot.kittybot.utils.MessageUtils;
+import de.kittybot.kittybot.utils.Utils;
+
+public class GuildSettings{
+
+	private final String guildId;
```
Comment (caneleex) [inline src/main/java/de/kittybot/kittybot/objects/GuildSettings.java:8]: why does this object need to hold the guild id? the only usage is pruning the cache with this stored id but for that refer to my previous comment

File: src/main/java/de/kittybot/kittybot/WebService.java (hunk 1 above)
Comment (TopiSenpai) [reply src/main/java/de/kittybot/kittybot/WebService.java:111]: Yeah true

File: src/main/java/de/kittybot/kittybot/cache/GuildSettingsCache.java (hunk 2 above)
Comment (TopiSenpai) [reply src/main/java/de/kittybot/kittybot/cache/GuildSettingsCache.java:161]: Yes yes blondy copied that from my old branch lol

File: src/main/java/de/kittybot/kittybot/objects/GuildSettings.java (hunk 3 above)
Comment (TopiSenpai) [reply src/main/java/de/kittybot/kittybot/objects/GuildSettings.java:8]: Yes yes me dumbo :)

**Pull request merged**